phabricatoremails/source.py

```diff
@@ -0,0 +1,63 @@
+import json
+from dataclasses import dataclass
+from typing import Optional
+
+import requests
+from requests import RequestException
+
+
+class PhabricatorException(Exception):
+    pass
+
+
+@dataclass
+class PhabricatorSource:
+    server: str
+    token: str
+    override_limit: Optional[int]
+
+    def _get(self, endpoint, **query_parameters):
+        query_parameters = {k: v for k, v in query_parameters.items() if v}
+        url = '{}/api/{}&api.token={}'.format(self.server, endpoint, self.token)
+        for key, value in query_parameters.items():
+            url += '&{}={}'.format(key, value)
+
+        try:
+            result = requests.get(url)
```

Comment:
rather than a GET request with the token in the url, use a POST (see moz-phab's `conduit.py`)